### DISCUSSION TITLE
install behind reverse proxy

### DISCUSSION THREAD

**Discussion by BGuiGui** [joplin-vieweb/joplin-vieweb#48] [Q&A]
Hello all,
Thanks for this nice web interface.
I tried to install it on my docker. I have nginx proxy manager as reverse proxy inside another docker network. This reverse proxy help me to access to all my services from outside.

So I think I only have to add theses Joplin containers to the both networks (Joplin and nginx). But I don't succeed to have it working. 

It seems that I can't get credentials from letsencrypt (waiting for certificate for my domain.com). I was thinking that I had issue with my network but not sure. I'm not sure to understand correctly the cert variables. I also tried on docker on laptop without reverse proxy and I have same problem.

Is ORIGIN variable need absolutely right domain and certificate is generate according to this domain ? So in localhost I could not run this app?

Thank you for your help !

**Comment by gri38** [comment]
Hi.
Thanks for your interest.
# About certbot
As you can see [here](https://github.com/joplin-vieweb/joplin-vieweb/#components):
the nginx and certbox containers are optional.

So if you already have a reverse nginx: use yours, and forget "mine", as explained [here](https://github.com/joplin-vieweb/joplin-vieweb/#installation): 
> If you have your own server, use [docker-compose-joplin-only.yml](https://github.com/joplin-vieweb/joplin-vieweb/blob/main/docker-compose-joplin-only.yml).

![image](https://github.com/joplin-vieweb/joplin-vieweb/assets/26554495/90c736e0-01f0-4a13-8df6-c50fdd367474)
# about origins
If you don't use my joplin-vieweb_nginx and joplin-vieweb-certbot (and you should use yours), then you don't need to set CERT_DOMAIN nor CERT_MAIL in x-common-variables.
You just need to set ORIGINS, as explained [here](https://github.com/joplin-vieweb/joplin-vieweb/#installation).
In origins you must set every "addresses" you expect your server will be accessed by.
For instance if you want to access it from localhost, and from your LAN (let's say your server is 192.168.1.50), and from outside with the domain name your_domain.com, then your docker-compose.yml will look like:
```yml
version: '3.4'

x-common-variables: &common-variables
   ORIGINS: "'https://your_domain.com', 'https://localhost', 'https://192.168.1.50'"

services:
  django-joplin-vieweb:
    image: gri38/django-joplin-vieweb:latest
...
```
Pay attention to the scheme (htt**p** or http**s**: it must match how you access your server).

Don't forget to replace PORT-TO-YOU-PROXY as explained in the install doc:

> 
>   - If you don't use nginx and certbot container (if you use docker-compose-joplin-only.yml):
>     - Change <PORT-TO-YOUR-PROXY> to fit your proxy setup.


# About certbot
If you have your own nginx reverse proxy: this one should handle certificate and http**s**. Do you already handle that for your other services?







Let me know if you manage to setup your servers' stuff. If not, explain and we can have a look to your docker-compose file.
